### DISCUSSION TITLE
Module not found: Can't resolve '@langfuse'

### DISCUSSION THREAD

**Discussion by christosuster** [langfuse/langfuse#3406] [Support] [answered]
Hi there, when I am trying to run Langfuse locally following this guide, [(https://github.com/langfuse/langfuse/blob/main/CONTRIBUTING.md)](url), I keep getting this error

@langfuse/ee:dev: src/env.ts:2:34 - error TS2307: Cannot find module '@langfuse/shared' or its corresponding type declarations.
@langfuse/ee:dev: 
@langfuse/ee:dev: 2 import { env as sharedEnv } from "@langfuse/shared";
@langfuse/ee:dev:                                    ~~~~~~~~~~~~~~~~~~
@langfuse/ee:dev: 
@langfuse/ee:dev: [5:34:45 AM] Found 1 error. Watching for file changes.
@langfuse/ee:dev: 
web:dev:   ▲ Next.js 14.2.6
web:dev:   - Local:        http://localhost:3000
web:dev:   - Experiments (use with caution):
web:dev:     · instrumentationHook
web:dev: 
web:dev:  ✓ Starting...
web:dev: Attention: Next.js now collects completely anonymous telemetry regarding usage.
web:dev: This information is used to shape Next.js' roadmap and prioritize features.
web:dev: You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
web:dev: https://nextjs.org/telemetry
web:dev: 
web:dev: warn  - It seems like you don't have a global error handler set up. It is recommended that you add a global-error.js file with Sentry instrumentation so that React rendering errors are reported to Sentry. Read more: https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#react-render-errors-in-app-router (you can suppress this warning by setting SENTRY_SUPPRESS_GLOBAL_ERROR_HANDLER_FILE_WARNING=1 as environment variable)
web:dev:  ○ Compiling /instrumentation ...
worker:dev: /workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:859
worker:dev:     return new TSError(diagnosticText, diagnosticCodes, diagnostics);
worker:dev:            ^
worker:dev: TSError: ⨯ Unable to compile TypeScript:
worker:dev: src/index.ts:4:24 - error TS2307: Cannot find module '@langfuse/shared/src/server' or its corresponding type declarations.
worker:dev: 
worker:dev: 4 import { logger } from "@langfuse/shared/src/server";
worker:dev:                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
worker:dev: 
worker:dev:     at createTSError (/workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:859:12)
worker:dev:     at reportTSError (/workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:863:19)
worker:dev:     at getOutput (/workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:1077:36)
worker:dev:     at Object.compile (/workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:1433:41)
worker:dev:     at Module.m._compile (/workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:1617:30)
worker:dev:     at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
worker:dev:     at Object.require.extensions.<computed> [as .ts] (/workspaces/langfuse/node_modules/.pnpm/ts-node@10.9.2_@types+node@20.11.29_typescript@5.4.5/node_modules/ts-node/src/index.ts:1621:12)
worker:dev:     at Module.load (node:internal/modules/cjs/loader:1208:32)
worker:dev:     at Function.Module._load (node:internal/modules/cjs/loader:1024:12)
worker:dev:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12) {
worker:dev:   diagnosticCodes: [ 2307 ]
worker:dev: }
worker:dev: [nodemon] app crashed - waiting for file changes before starting...
web:dev:  ⨯ ./src/datadog.server.config.ts:11:1
web:dev: Module not found: Can't resolve '@langfuse/shared/src/server'
web:dev:    9 | // import { BullMQInstrumentation } from "@appsignal/opentelemetry-instrumentation-bullmq";
web:dev:   10 | import { UndiciInstrumentation } from "@opentelemetry/instrumentation-undici";
web:dev: > 11 | import { logger } from "@langfuse/shared/src/server";
web:dev:      | ^
web:dev:   12 | import { WinstonInstrumentation } from "@opentelemetry/instrumentation-winston";
web:dev:   13 |
web:dev:   14 | if (!process.env.VERCEL && process.env.NEXT_PUBLIC_LANGFUSE_CLOUD_REGION) {
web:dev: 
web:dev: https://nextjs.org/docs/messages/module-not-found
web:dev: 
web:dev: Import trace for requested module:
web:dev: ./src/instrumentation.ts
web:dev: Error: An error occurred while loading instrumentation hook: Cannot find module '/workspaces/langfuse/web/.next/server/instrumentation'
web:dev: Require stack:
web:dev: - /workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/dev/next-dev-server.js
web:dev: - /workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/next.js
web:dev: - /workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/lib/start-server.js
web:dev:     at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
web:dev:     at /workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/require-hook.js:55:36
web:dev:     at Module._load (node:internal/modules/cjs/loader:986:27)
web:dev:     at Module.require (node:internal/modules/cjs/loader:1233:19)
web:dev:     at mod.require (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/require-hook.js:65:28)
web:dev:     at require (node:internal/modules/helpers:179:18)
web:dev:     at DevServer.runInstrumentationHookIfAvailable (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/dev/next-dev-server.js:436:51)
web:dev:     at async Span.traceAsyncFn (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/trace/trace.js:154:20)
web:dev:     at async DevServer.prepareImpl (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/dev/next-dev-server.js:214:9)
web:dev:     at async NextServer.prepare (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/next.js:161:13)
web:dev:     at async initializeImpl (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/lib/render-server.js:98:5)
web:dev:     at async initialize (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/lib/router-server.js:423:22)
web:dev:     at async Server.<anonymous> (/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/lib/start-server.js:249:36) {
web:dev:   code: 'MODULE_NOT_FOUND',
web:dev:   requireStack: [
web:dev:     '/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/dev/next-dev-server.js',
web:dev:     '/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/next.js',
web:dev:     '/workspaces/langfuse/node_modules/.pnpm/next@14.2.6_@babel+core@7.24.3_@opentelemetry+api@1.8.0_@playwright+test@1.43.1_babel-plugin-_avqftbgnxaksteqelfvq3xgacu/node_modules/next/dist/server/lib/start-server.js'
web:dev:   ]
web:dev: }

It seems like @langfuse cannot be found. I tried running it on Windows, WSL2 and even on Codespace but I can't seem to make it run. Am I doing something wrong here? Please help me out. Thanks.

**Comment by marcklingen** [comment]
Did you follow exactly these steps using pnpm on node 20?
https://github.com/langfuse/langfuse/blob/main/CONTRIBUTING.md#development-setup

**Reply by christosuster** [reply, marked as answer]
Yes I did. In the end I found that the solution to this specific problem was to increase --max-old-space-size to a larger number. My system doesn't have enough ram so I ran it on Github Codespaces and it built successfully. However, even after following all the instructions, everything runs except for my frontend. In Codespaces I can open the port 3000 link and see it loading for a while before it shuts down. All the other services still remain running though.

![image](https://github.com/user-attachments/assets/e1a3b7d7-ce6d-47b1-ab08-07d7b85c0b58)

**Reply by marcklingen** [reply]
Thanks for sharing. It makes sense that the overall application might exhaust memory if little is available as some is already used by the different infrastructure components run via docker.

**Reply by a-imran99** [reply]
Hi @christosuster, Facing the same issue, everything runs except the UI, same error. 
(worker:dev: src/index.ts(4,24): error TS2307: Cannot find module '@langfuse/shared/src/server' or its corresponding type declarations.)

**Reply by christosuster** [reply]
> Hi @christosuster, Facing the same issue, everything runs except the UI, same error. (worker:dev: src/index.ts(4,24): error TS2307: Cannot find module '@langfuse/shared/src/server' or its corresponding type declarations.)

Hi @a-imran99, if your reason for facing this issue as same as mine, then this is definitely a memory issue. For me, I had to get a RAM upgrade to 32 GB to make everything run smoothly. When you run `pnpm run dev`, the frontend is built and that built file is later referred to using `@langfuse`. So because the system is not getting sufficient memory, the build fails and in turn `@langfuse` module cannot be referred to because it couldn't be built. So you need to make sure you have sufficient memory and then increase the max-old-space-size of your node application to a large number. I set it to 12288 for my case. You can find information on how it is done over here [https://stackoverflow.com/questions/56982005/where-do-i-set-node-options-max-old-space-size-2048](url). Hope this fixes your issue.
